phoenix-queryserver/src/main/java/org/apache/phoenix/queryserver/server/QueryServer.java

```diff
@@ -595,6 +595,15 @@ public <T> T doAsRemoteUser(String remoteUserName, String remoteAddress,
 
       // Proxy this user on top of the server's user (the real user). Get a cached instance, the
       // LoadingCache will create a new instance for us if one isn't cached.
+
+      // realm got removed from remoteUserName in CALCITE-4152
+      // so we remove the instance name to avoid geting KerberosName$NoMatchingRule exception
+
+      int separatorIndex = remoteUserName.indexOf('/');
```

Comment:
I don't recall anymore if there's a chance that `remoteUserName` might also contain an `@REALM`. You could be defensive and guard against that for "free" (do a second `indexOf` and `substring` block)

Reply:
As discussed offline, this was a bad idea, we don't want to add Hadoop dependencies back to the PQS.

Reply:
oh, that's a very good idea... I mean, unless you enjoy having an entire continent of extra transient dependencies ;)